ksqldb-common/src/main/java/io/confluent/ksql/function/GenericsUtil.java

```diff
@@ -213,23 +221,29 @@ private static boolean resolveGenerics(
     if (schema instanceof LambdaType) {
       final LambdaType lambdaType = (LambdaType) schema;
       final SqlLambda sqlLambda = instance.getSqlLambdaOrThrow();
-      if (sqlLambda.getInputType().size() != lambdaType.inputTypes().size()) {
+      if (lambdaType.inputTypes().size() == sqlLambda.getNumInputs()) {
+        if (resolveLambdaType) {
+          int i = 0;
+          for (final ParamType paramType : lambdaType.inputTypes()) {
+            if (!resolveGenerics(mapping, paramType,
+                SqlArgument.of(sqlLambda.getInputType().get(i)), resolveLambdaType)) {
+              return false;
+            }
+            i++;
+          }
+          return resolveGenerics(
+              mapping,
+              lambdaType.returnType(),
+              SqlArgument.of(sqlLambda.getReturnType()),
+              resolveLambdaType
+          );
+        } else {
+          return true;
+        }
+      } else {
```

Comment:
You're right, I changed this back to returning false.

Reply:
For my own education, it seems that if users indeed register two function as above, then 1) `UdfIndex#getCandidates` would return more than one candidate, and 2) we have to rely on `UdfIndex#Node#compare` to pick the right one, which would be relying on `order` since `int compare = compareFunctions.compare(value, other.value)` would return zero.

Does that mean, that users declaring these functions under the same class should always declare the first and then the second so that `transform(list, (x, i) -> 'foo')` has a higher order? I.e. do we enforce a declaration ordering for cases like above?

Reply:
I don't think the ordering would matter, in Almog's example, they lambda has different input types so there shouldn't be confusion when verifying if the variation the user is trying to use matches one of the candidates, it should only be possible to return one of them

Reply:
Got it, so in what scenarios the `UdfIndex#getCandidates` could return multiple results?

Reply:
I think the only scenario would be the second getCandidates call
```
    // if none were found (candidates is empty) try again with
    // implicit casting
    getCandidates(arguments, 0, root, candidates, new HashMap<>(), true);
```
when we allow casting of the function arguments when we do `return ParamTypes.areCompatible(argument, type, allowCasts);` in `accepts()`